goalc/compiler/compilation/Asm.cpp

```diff
@@ -274,15 +276,15 @@ Val* Compiler::compile_asm_lvf(const goos::Object& form, const goos::Object& res
     throw_compiler_error(form, "Cannot .lvf into this. Got a {}.", dest->print());
   }
   auto src = compile_error_guard(args.unnamed.at(1), env);
-  auto as_co = dynamic_cast<MemoryOffsetConstantVal*>(src);
   auto as_sv = dynamic_cast<StaticVal*>(src);
   MemLoadInfo info;
   info.sign_extend = false;
   info.size = 16;
   info.reg = RegClass::VECTOR_FLOAT;
-  if (as_co) {
+  if (args.has_named("offset")) {
+    int offset = args.named.at("offset").as_int();
     // can do a clever offset here
-    env->emit_ir<IR_LoadConstOffset>(dest, as_co->offset, as_co->base->to_gpr(env), info, color);
+    env->emit_ir<IR_LoadConstOffset>(dest, 0, src->to_gpr(env), info, color);
```

Comment:
I think this isn't quite right. The `MemoryOffsetConstantVal` represents a memory location plus some constant offset.  We want to propagate constant memory offsets at compile to avoid using temporary registers and addition instructions when possible.  

There should be 6 cases:
- `as_co` and no `:offset` - use the old `as_co->offset` and `as_co->base`
- `as_co` and `:offset` - use `as_co->offset + offset` and `as_co->base`
- No `as_co` and no `:offset` - use `0` and `src->to_gpr(env)`
- No `as_co` and `:offset` - use `offset` and `src->to_gpr(env)`
- `as_sv` and no `:offset` - do a `IR_StaticVarLoad`
- `as_sv` and `:offset` - throw a compiler error.  Loading directly from a `(new 'static ...)` isn't something that the original GOAL supported - I only added it to make it faster to load constant vectors/floats.